src/Symfony/Component/DependencyInjection/ParameterBag/EnvPlaceholderParameterBag.php

```diff
@@ -79,6 +79,13 @@ public function mergeEnvPlaceholders(self $bag)
                 $this->envPlaceholders[$env] += $placeholders;
             }
         }
+        foreach ($bag->getProvidedTypes() as $prefix => $types) {
```

Comment:
I think this file doesn't need to be modified: merging happens after setProvidedTypes is called, and should not happen anymore after

Reply:
They get lost in `MergeExtensionConfigurationParameterBag::__construct` as we overwrite `$resolvingBag`.

But agree; this method is called `mergeEnvPlaceholders`.. so doesnt imply it merges types. Now fixed.